code/go/0chain.net/smartcontract/dbs/event/process.go

```diff
@@ -206,45 +206,35 @@ func mergeEvents(round int64, block string, events []Event) ([]Event, error) {
 func (edb *EventDb) addEventsWorker(ctx context.Context) {
 	for {
 		es := <-edb.eventsChannel
-		edb.addEvents(ctx, es)
+
+		tx, err := edb.Begin()
+		if err != nil {
+			logging.Logger.Error("error starting transaction", zap.Error(err))
+		}
+
+		tx.addEvents(ctx, es)
 		tse := time.Now()
 		tags := make([]int, 0, len(es.events))
 		for _, event := range es.events {
-			var err error = nil
-			switch EventType(event.Type) {
-			case TypeStats:
-				tags = append(tags, event.Tag)
-				ts := time.Now()
-				err = edb.addStat(event)
-				du := time.Since(ts)
-				if du.Milliseconds() > 50 {
-					logging.Logger.Warn("event db save slow - addStat",
-						zap.Any("duration", du),
-						zap.Int("event tag", event.Tag),
-						zap.Int64("round", es.round),
-						zap.String("block", es.block),
-						zap.Int("block size", es.blockSize),
-					)
-				}
-			case TypeError:
-				err = edb.addError(Error{
-					TransactionID: event.TxHash,
-					Error:         fmt.Sprintf("%v", event.Data),
-				})
-
-			default:
-			}
+			tx.Get().SavePoint(event.Index)
```

Comment:
event.Index is not global unique, it could be blobber id for example, so not sure which point the rollbackTo will go to if we have multiple saved points with the same name.